src/components/input/input.scss

```diff
@@ -33,6 +39,15 @@ $md-input-underline-disabled-background-image: linear-gradient(to right,
   }
 }
 
+/**
+ * Applies a floating placeholder above the input before chrome's and safari's auto-fill commits.
+ * Firefox has selector issues for :-webkit-autofill, thats why this mixin is duplicated and is used seperately below 
+ */
+%md-input-placeholder-floating-autofill {
+  visibility: visible;
+  padding-bottom: 5px;
+  transform: translateY(-100%) scale(0.75);
+}
```

Comment:
@jelbourn I'm not sure that I clearly understand your note. If you suggest to make **one **mixin, than it means to **rollback ** change, because it **WAS** one mixin which doesn't work. **Why?** It generates joined css, something like this: `.md-input-placeholder.md-float:not(.md-empty), .md-input-placeholder.md-float.md-focused, input:-webkit-autofill + .md-input-placeholder` which is not selected by FF. Why? **input:-webkit-autofill** css is invalid for FF and it ignores all joined styles.

How to improve? Well, if we split it in two different groups of css `input:-webkit-autofill` and others, than it works.

Only way i found to create these two groups in rendered css is to make another mixin. That's why it is 'duplicated'. May be there is some other option to force scss compiler render different groups in css when we use 'extend' of mixins?